diff --git a/backend/meson.build b/backend/meson.build
index 14629f2..2f972a8 100644
--- a/backend/meson.build
+++ b/backend/meson.build
@@ -53,7 +53,6 @@ foreach backend, backend_mime_types: backends
   )
 
   i18n.merge_file(
-    appstream,
     input: appstream_in,
     output: appstream,
     po_dir: po_dir,
diff --git a/data/meson.build b/data/meson.build
index 8a308b8..afc3020 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -15,7 +15,6 @@ foreach desktop: desktops
   )
 
   i18n.merge_file(
-    desktop,
     type: 'desktop',
     input: desktop_in,
     output: desktop,
diff --git a/meson.build b/meson.build
index 07fb8ec..0bb5d0f 100644
--- a/meson.build
+++ b/meson.build
@@ -498,7 +498,6 @@ install_headers(
 appdata = ev_namespace + '.appdata.xml'
 
 i18n.merge_file(
-  appdata,
   input: appdata + '.in',
   output: appdata,
   po_dir: po_dir,
