diff --git a/glib/gspawn.c b/glib/gspawn.c
index 0a2cbe54a..a2ea9b9bc 100644
--- a/glib/gspawn.c
+++ b/glib/gspawn.c
@@ -1757,14 +1757,16 @@ do_exec (gint                  child_err_report_fd,
           if (safe_dup2 (child_err_report_fd, 3) < 0)
             write_err_and_exit (child_err_report_fd, CHILD_DUP2_FAILED);
           set_cloexec (GINT_TO_POINTER (0), 3);
-          if (safe_closefrom (4) < 0)
-            write_err_and_exit (child_err_report_fd, CHILD_CLOSE_FAILED);
+          //if (safe_closefrom (4) < 0)
+          // write_err_and_exit (child_err_report_fd, CHILD_CLOSE_FAILED);
+          safe_closefrom (4);
           child_err_report_fd = 3;
         }
       else
         {
-          if (safe_fdwalk_set_cloexec (3) < 0)
-            write_err_and_exit (child_err_report_fd, CHILD_CLOSE_FAILED);
+          //if (safe_fdwalk_set_cloexec (3) < 0)
+          // write_err_and_exit (child_err_report_fd, CHILD_CLOSE_FAILED);
+          safe_fdwalk (set_cloexec, GINT_TO_POINTER (3));
         }
     }
   else
