diff --git a/Makefile b/Makefile
index 599b145..22c4a8b 100644
--- a/Makefile
+++ b/Makefile
@@ -7,20 +7,22 @@ SPECFILE	:= keyutils.spec
 NO_GLIBC_KEYERR	:= 0
 NO_ARLIB	:= 0
 NO_SOLIB	:= 0
-ETCDIR		:= /etc
-BINDIR		:= /bin
-SBINDIR		:= /sbin
-SHAREDIR	:= /usr/share/keyutils
-MANDIR		:= /usr/share/man
+YDFSDIR	= $(HOME)/$(ARCH)
+LIBDIR	= $(YDFSDIR)/lib
+ETCDIR		:= $(YDFSDIR)/etc
+BINDIR		:= $(YDFSDIR)/bin
+SBINDIR		:= $(YDFSDIR)/sbin
+SHAREDIR	:= $(YDFSDIR)/share/keyutils
+MANDIR		:= $(YDFSDIR)/share/man
 MAN1		:= $(MANDIR)/man1
 MAN3		:= $(MANDIR)/man3
 MAN5		:= $(MANDIR)/man5
 MAN7		:= $(MANDIR)/man7
 MAN8		:= $(MANDIR)/man8
-INCLUDEDIR	:= /usr/include
+INCLUDEDIR	:= $(YDFSDIR)/include
 LN		:= ln
 LNS		:= $(LN) -sf
-PREFIX 		:= /usr
+PREFIX 		:= $(YDFSDIR)
 
 ###############################################################################
 #
@@ -183,6 +185,7 @@ cxx: cxx.stamp
 # Install everything
 #
 ###############################################################################
+USRLIBDIR	= $(HOME)/$(ARCH)/lib
 pkgconfig:
 	sed \
 	-e 's,@VERSION\@,$(VERSION),g' \
@@ -285,13 +288,6 @@ SRCBALL	:= rpmbuild/SOURCES/$(TARBALL)
 ZSRCBALL := rpmbuild/SOURCES/$(ZTARBALL)
 
 BUILDID	:= .local
-rpmver0	:= $(shell rpmspec -q ./keyutils.spec --define "buildid $(BUILDID)")
-rpmver1	:= $(word 1,$(rpmver0))
-rpmver2	:= $(subst ., ,$(rpmver1))
-rpmver3	:= $(lastword $(rpmver2))
-rpmver4	:= $(patsubst %.$(rpmver3),%,$(rpmver1))
-rpmver	:= $(patsubst keyutils-%,%,$(rpmver4))
-SRPM	:= rpmbuild/SRPMS/keyutils-$(rpmver).src.rpm
 
 RPMBUILDDIRS := \
 	--define "_srcrpmdir $(CURDIR)/rpmbuild/SRPMS" \
