diff --git a/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx b/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
index 79a3780..2cc27e5 100644
--- a/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
+++ b/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
@@ -202,6 +202,9 @@ type_info * RTTI::getRTTI( typelib_CompoundTypeDescription *pTypeDescr )
 extern "C" {
 static void _GLIBCXX_CDTOR_CALLABI deleteException( void * pExc )
 {
+    /*
+     * FIX https://github.com/yledoare/ydfs/issues/1 
+     *
     __cxa_exception const * header = static_cast<__cxa_exception const *>(pExc) - 1;
     typelib_TypeDescription * pTD = nullptr;
     OUString unoName( toUNOname( header->exceptionType->name() ) );
@@ -212,6 +215,7 @@ static void _GLIBCXX_CDTOR_CALLABI deleteException( void * pExc )
         ::uno_destructData( pExc, pTD, cpp_release );
         ::typelib_typedescription_release( pTD );
     }
+    */
 }
 }
 
