diff --git a/include/plist/plist.h b/include/plist/plist.h
index ab91612..de02304 100644
--- a/include/plist/plist.h
+++ b/include/plist/plist.h
@@ -29,20 +29,7 @@ extern "C"
 {
 #endif
 
-#if _MSC_VER && _MSC_VER < 1700
-    typedef __int8 int8_t;
-    typedef __int16 int16_t;
-    typedef __int32 int32_t;
-    typedef __int64 int64_t;
-
-    typedef unsigned __int8 uint8_t;
-    typedef unsigned __int16 uint16_t;
-    typedef unsigned __int32 uint32_t;
-    typedef unsigned __int64 uint64_t;
-
-#else
 #include <stdint.h>
-#endif
 
 #ifdef __llvm__
   #if defined(__has_extension)
