diff --git a/CMakeLists.txt b/CMakeLists.txt
index cabae8c..4f59c65 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -301,11 +301,6 @@ if(NOT WIN32)
 	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} ${EXTRA_FLAGS_CONFIG} ${EXTRA_FLAGS_RELEASE}" CACHE STRING "Release build flags" FORCE)
 	set(CMAKE_EXE_LINKER_FLAGS_RELEASE "" CACHE STRING "" FORCE)
 # set the arch to use for Release build if provided
-	if(ARCH)
-		MESSAGE("adding -march=${ARCH} to Release build")
-		set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=${ARCH}" CACHE STRING "Release build flags" FORCE)
-		set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=${ARCH}" CACHE STRING "Release build flags" FORCE)
-	endif()
 
 # PGO and LTO for GCC
 	if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
