diff --git a/src/compat-api.h b/src/compat-api.h
index fde2f4b..8a1fcf9 100644
--- a/src/compat-api.h
+++ b/src/compat-api.h
@@ -102,4 +102,8 @@
 
 #endif
 
+#if ABI_VIDEODRV_VERSION < SET_ABI_VERSION(25, 2)
+#define secondary_dst slave_dst
+#endif
+
 #endif
diff --git a/src/nouveau_exa.c b/src/nouveau_exa.c
index 55df6f8..db3b112 100644
--- a/src/nouveau_exa.c
+++ b/src/nouveau_exa.c
@@ -157,7 +157,7 @@ nouveau_exa_destroy_pixmap(ScreenPtr pScreen, void *priv)
 
 #ifdef NOUVEAU_PIXMAP_SHARING
 static Bool
-nouveau_exa_share_pixmap_backing(PixmapPtr ppix, ScreenPtr slave, void **handle_p)
+nouveau_exa_share_pixmap_backing(PixmapPtr ppix, ScreenPtr secondary, void **handle_p)
 {
 	struct nouveau_bo *bo = nouveau_pixmap_bo(ppix);
 	struct nouveau_pixmap *nvpix = nouveau_pixmap(ppix);
diff --git a/src/nv_driver.c b/src/nv_driver.c
index e72a6b6..f9ab4af 100644
--- a/src/nv_driver.c
+++ b/src/nv_driver.c
@@ -559,16 +559,16 @@ redisplay_dirty(ScreenPtr screen, PixmapDirtyUpdatePtr dirty)
 {
 	RegionRec pixregion;
 
-	PixmapRegionInit(&pixregion, dirty->slave_dst);
+	PixmapRegionInit(&pixregion, dirty->secondary_dst);
 
-	DamageRegionAppend(&dirty->slave_dst->drawable, &pixregion);
+	DamageRegionAppend(&dirty->secondary_dst->drawable, &pixregion);
 #ifdef HAS_DIRTYTRACKING_ROTATION
 	PixmapSyncDirtyHelper(dirty);
 #else
 	PixmapSyncDirtyHelper(dirty, &pixregion);
 #endif
 
-	DamageRegionProcessPending(&dirty->slave_dst->drawable);
+	DamageRegionProcessPending(&dirty->secondary_dst->drawable);
 	RegionUninit(&pixregion);
 }
 
diff --git a/ydfs-error b/ydfs-error
index e69de29..07ca764 100644
--- a/ydfs-error
+++ b/ydfs-error
@@ -0,0 +1,29 @@
+--2022-02-17 09:23:50--  https://raw.githubusercontent.com/archlinux/svntogit-packages/packages//xf86-video-nouveau/trunk/https://xorg.freedesktop.org/archive/individual/driver/xf86-video-nouveau-1.0.17.tar.bz2.sig
+SSL_INIT
+Résolution de raw.githubusercontent.com… 185.199.111.133, 185.199.108.133, 185.199.109.133, ...
+Connexion à raw.githubusercontent.com|185.199.111.133|:443… connecté.
+requête HTTP transmise, en attente de la réponse… 301 Moved Permanently
+Emplacement : /archlinux/svntogit-packages/packages/xf86-video-nouveau/trunk/https:/xorg.freedesktop.org/archive/individual/driver/xf86-video-nouveau-1.0.17.tar.bz2.sig [suivant]
+--2022-02-17 09:23:51--  https://raw.githubusercontent.com/archlinux/svntogit-packages/packages/xf86-video-nouveau/trunk/https:/xorg.freedesktop.org/archive/individual/driver/xf86-video-nouveau-1.0.17.tar.bz2.sig
+SSL_INIT
+Réutilisation de la connexion existante à raw.githubusercontent.com:443.
+requête HTTP transmise, en attente de la réponse… 404 Not Found
+2022-02-17 09:23:51 erreur 404 : Not Found.
+
+--2022-02-17 09:23:52--  https://raw.githubusercontent.com/archlinux/svntogit-packages/packages//xf86-video-nouveau/trunk/xorg-server-21.1.diff
+SSL_INIT
+Résolution de raw.githubusercontent.com… 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
+Connexion à raw.githubusercontent.com|185.199.108.133|:443… connecté.
+requête HTTP transmise, en attente de la réponse… 301 Moved Permanently
+Emplacement : /archlinux/svntogit-packages/packages/xf86-video-nouveau/trunk/xorg-server-21.1.diff [suivant]
+--2022-02-17 09:23:53--  https://raw.githubusercontent.com/archlinux/svntogit-packages/packages/xf86-video-nouveau/trunk/xorg-server-21.1.diff
+SSL_INIT
+Réutilisation de la connexion existante à raw.githubusercontent.com:443.
+requête HTTP transmise, en attente de la réponse… 200 OK
+Taille : 1743 (1,7K) [text/plain]
+Sauvegarde en : « patch »
+
+     0K .                                                     100% 2,72M=0,001s
+
+2022-02-17 09:23:53 (2,72 MB/s) — « patch » sauvegardé [1743/1743]
+
diff --git a/ydfs-log b/ydfs-log
index 53b8449..69a6fc8 100644
--- a/ydfs-log
+++ b/ydfs-log
@@ -1,3 +1,43 @@
 Package source : xf86-video-nouveau-1.0.17 CC : /usr/bin/gcc, PREFIX : /home/linuxconsole2022/x86/
 /home/linuxconsole2022/ydfs/src/xf86-video-nouveau-1.0.17
 Dépôt Git vide initialisé dans /home/linuxconsole2022/ydfs/src/xf86-video-nouveau-1.0.17/.git/
+https://raw.githubusercontent.com/archlinux/svntogit-packages/packages//xf86-video-nouveau/trunk/https://xorg.freedesktop.org/archive/individual/driver/xf86-video-nouveau-1.0.17.tar.bz2.sig
+https://raw.githubusercontent.com/archlinux/svntogit-packages/packages//xf86-video-nouveau/trunk/xorg-server-21.1.diff
+https://raw.githubusercontent.com/archlinux/svntogit-packages/packages//xf86-video-nouveau/trunk/https://xorg.freedesktop.org/archive/individual/driver/xf86-video-nouveau-1.0.17.tar.bz2.sig
+https://raw.githubusercontent.com/archlinux/svntogit-packages/packages//xf86-video-nouveau/trunk/xorg-server-21.1.diff
+can't find file to patch at input line 5
+Perhaps you used the wrong -p or --strip option?
+The text leading up to this was:
+--------------------------
+|diff --git a/src/compat-api.h b/src/compat-api.h
+|index fde2f4b1cfde75875c07bfe13524dc6ba2661382..8a1fcf9be1c5d1ceb48a50f2ed533d93ec7ff4c7 100644
+|--- a/src/compat-api.h
+|+++ b/src/compat-api.h
+--------------------------
+No file to patch.  Skipping patch.
+1 out of 1 hunk ignored
+can't find file to patch at input line 18
+Perhaps you used the wrong -p or --strip option?
+The text leading up to this was:
+--------------------------
+|diff --git a/src/nouveau_exa.c b/src/nouveau_exa.c
+|index 55df6f8f11c9e14b1891e5c841faef10c17f0a35..db3b112a2db70f8e902e54aa3af99e51e7d0c6f7 100644
+|--- a/src/nouveau_exa.c
+|+++ b/src/nouveau_exa.c
+--------------------------
+No file to patch.  Skipping patch.
+1 out of 1 hunk ignored
+can't find file to patch at input line 31
+Perhaps you used the wrong -p or --strip option?
+The text leading up to this was:
+--------------------------
+|diff --git a/src/nv_driver.c b/src/nv_driver.c
+|index e72a6b65a81119f12f3608295a4547762a866ad7..f9ab4af19361d99d74f580d1ff3f28d49843e8b0 100644
+|--- a/src/nv_driver.c
+|+++ b/src/nv_driver.c
+--------------------------
+No file to patch.  Skipping patch.
+1 out of 1 hunk ignored
+patching file src/compat-api.h
+patching file src/nouveau_exa.c
+patching file src/nv_driver.c
