diff --git a/js/src/Makefile.in b/js/src/Makefile.in
index 9c58960..0286300 100644
--- a/js/src/Makefile.in
+++ b/js/src/Makefile.in
@@ -661,9 +661,6 @@ EXTRA_LIBS += $(MOZ_ZLIB_LIBS)
 # Enable zlib usage if zlib has been located. When building the browser on
 # Windows, MOZ_ZLIB_LIBS is empty because zlib is part of libmozglue. We thus
 # also enable zlib if mozglue is present.
-ifneq (,$(MOZ_ZLIB_LIBS)$(MOZ_GLUE_LDFLAGS))
-DEFINES += -DUSE_ZLIB
-endif
 
 # Prevent floating point errors caused by VC++ optimizations
 ifdef _MSC_VER
