diff --git a/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/util/convert.cpp b/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/util/convert.cpp
index 97bfe91..cdba940 100644
--- a/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/util/convert.cpp
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/util/convert.cpp
@@ -3,7 +3,7 @@
 namespace mbgl {
 namespace util {
 
-template std::array<float, 2> convert(const std::array<int32_t, 2>&);
+//template std::array<float, 2> convert(const std::array<int32_t, 2>&);
 
 } // namespace util
 } // namespace mbgl
