diff --git a/networking/udhcp/dhcpc.c b/networking/udhcp/dhcpc.c
index 739870b..d73fc09 100644
--- a/networking/udhcp/dhcpc.c
+++ b/networking/udhcp/dhcpc.c
@@ -730,7 +730,7 @@ static NOINLINE int send_discover(uint32_t xid, uint32_t requested)
 	 */
 	add_client_options(&packet);
 
-	bb_info_msg("sending %s", "discover");
+	//bb_info_msg("sending %s", "discover");
 	return raw_bcast_from_client_data_ifindex(&packet, INADDR_ANY);
 }
 
