diff --git a/common/mp4ff/Makefile.am b/common/mp4ff/Makefile.am
index 3c495dc..438db81 100644
--- a/common/mp4ff/Makefile.am
+++ b/common/mp4ff/Makefile.am
@@ -1,7 +1,8 @@
-lib_LIBRARIES = libmp4ff.a
-include_HEADERS = mp4ff.h mp4ffint.h
+lib_LTLIBRARIES = libmp4ff.la
+include_HEADERS = mp4ff.h mp4ffint.h mp4ff_int_types.h
 
-libmp4ff_a_CFLAGS = -DUSE_TAGGING=1
+AM_CFLAGS = -DUSE_TAGGING=1
 
-libmp4ff_a_SOURCES = mp4ff.c mp4atom.c mp4meta.c mp4sample.c mp4util.c \
+libmp4ff_la_LDFLAGS = -version-info 0:0:0
+libmp4ff_la_SOURCES = mp4ff.c mp4atom.c mp4meta.c mp4sample.c mp4util.c \
 		     mp4tagupdate.c mp4ff.h mp4ffint.h mp4ff_int_types.h
diff --git a/frontend/Makefile.am b/frontend/Makefile.am
index 74af733..3c40526 100644
--- a/frontend/Makefile.am
+++ b/frontend/Makefile.am
@@ -1,11 +1,11 @@
 bin_PROGRAMS = faad
-man_MANS = faad.man
+man1_MANS = faad.man
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/common/faad \
 	   -I$(top_srcdir)/common/mp4ff
 
 faad_LDADD = $(top_builddir)/libfaad/libfaad.la \
-	     $(top_builddir)/common/mp4ff/libmp4ff.a
+	     $(top_builddir)/common/mp4ff/libmp4ff.la
 
 faad_SOURCES = main.c \
 	       audio.c audio.h \
