diff --git a/glib/gmain.c b/glib/gmain.c
index ee23a65..cb33183 100644
--- a/glib/gmain.c
+++ b/glib/gmain.c
@@ -2935,10 +2935,10 @@ g_get_monotonic_time (void)
   struct timespec ts;
   gint result;
 
-  result = clock_gettime (CLOCK_MONOTONIC, &ts);
+  result = clock_gettime (CLOCK_REALTIME, &ts);
 
-  if G_UNLIKELY (result != 0)
-    g_error ("GLib requires working CLOCK_MONOTONIC");
+  //if G_UNLIKELY (result != 0)
+   // g_error ("GLib requires working CLOCK_MONOTONIC, result = %d",result);
 
   return (((gint64) ts.tv_sec) * 1000000) + (ts.tv_nsec / 1000);
 }
