diff --git a/Makefile b/Makefile
index 475d522..e18ccfc 100644
--- a/Makefile
+++ b/Makefile
@@ -76,7 +76,7 @@ PREFIX=/usr/local
 endif
 
 ifndef DIR_INSTALL
-DIR_INSTALL=$(PREFIX)/lib/jxrlib-$(JXR_VERSION)
+DIR_INSTALL=$(PREFIX)
 endif
 
 CD=cd
@@ -249,17 +249,6 @@ $(DIR_BUILD)/libjxr++.$(LIBSUFFIX): $(OBJ_CXX) | $(LIBRARIES)
 ## Java Wrapper library
 ##
 
-SRC_JAVA=$(wildcard $(DIR_SRC)/$(DIR_JAVA)/$(JAVA_PKG)/*.java)
-
-swig:
-	mkdir -p $(DIR_SRC)/$(DIR_JAVA)/$(JAVA_PKG)
-	$(SWIG) -java -c++ -package ome.jxrlib -outdir $(DIR_SRC)/$(DIR_JAVA)/$(JAVA_PKG) -o $(DIR_SRC)/$(DIR_JAVA)/JXR_wrap.cxx java/JXR.i
-
-$(DIR_BUILD)/libjxrjava.$(LIBSUFFIX): $(LIBRARIES) $(CXX_LIBRARIES)
-	@echo "Building JNI"
-	@echo "JAVA_INCLUDE=$(JAVA_INCLUDE)"
-	$(CXX) -o $(DIR_BUILD)/libjxrjava.$(LIBSUFFIX) -shared -I$(JAVA_INCLUDE) -I$(JAVA_INCLUDE)/$(PLATFORM) -I$(DIR_CXX)/lib $(CXXFLAGS) $(OBJ_SYS) $(OBJ_ENC) $(OBJ_DEC) $(OBJ_GLUE) $(OBJ_TEST) $(OBJ_CXX) $(DIR_JAVA)/JXR_wrap.cxx
-
 ##--------------------------------
 ##
 ## Enc app files
@@ -299,7 +288,7 @@ $(DIR_BUILD)/$(CXXDECAPP): $(DIR_SRC)/$(DIR_CXX)/$(CXXDECAPP).cpp $(LIBRARIES) $
 ##
 ## JPEG XR library
 ##
-all: $(DIR_BUILD)/$(ENCAPP) $(DIR_BUILD)/$(DECAPP) $(DIR_BUILD)/$(CXXDECAPP) $(LIBRARIES) $(CXX_LIBRARIES) $(DIR_BUILD)/libjxrjava.$(LIBSUFFIX)
+all: $(DIR_BUILD)/$(ENCAPP) $(DIR_BUILD)/$(DECAPP) $(DIR_BUILD)/$(CXXDECAPP) $(LIBRARIES) $(CXX_LIBRARIES)
 
 clean:
 	rm -rf $(DIR_BUILD)/*App $(DIR_BUILD)/**/*.o $(DIR_BUILD)/**/*.class $(DIR_BUILD)/libj*.a $(DIR_BUILD)/libj*.$(LIBSUFFIX) $(DIR_BUILD)/libjxr.pc $(DIR_BUILD)/$(CXXDECAPP) $(DIR_BUILD)/$(JAR)
diff --git a/libjxr.pc.in b/libjxr.pc.in
index b581451..90daa0e 100644
--- a/libjxr.pc.in
+++ b/libjxr.pc.in
@@ -9,4 +9,4 @@ Description: A library for reading JPEG XR images.
 Version: %(JXR_VERSION)s
 Libs: -L${libdir} -ljpegxr -ljxrglue
 Libs.private: -lm 
-Cflags: -I${includedir}/libjxr/common -I${includedir}/libjxr/image/x86 -I${includedir}/libjxr/image -I${includedir}/libjxr/glue -I${includedir}/libjxr/test -D__ANSI__ -DDISABLE_PERF_MEASUREMENT %(JXR_ENDIAN)s
+Cflags: -I${includedir}/jxrlib
