diff --git a/src/org.mate.terminal.gschema.xml.in b/src/org.mate.terminal.gschema.xml.in
index 2ab3d7e..933bdb6 100644
--- a/src/org.mate.terminal.gschema.xml.in
+++ b/src/org.mate.terminal.gschema.xml.in
@@ -99,12 +99,12 @@
       <description>True if the menubar should be shown in new windows, for windows/tabs with this profile.</description>
     </key>
     <key name="foreground-color" type="s">
-      <default>'#000000'</default>
+      <default>'#FFFFFF'</default>
       <summary>Default color of text in the terminal</summary>
       <description>Default color of text in the terminal, as a color specification (can be HTML-style hex digits, or a color name such as "red").</description>
     </key>
     <key name="background-color" type="s">
-      <default>'#FFFFDD'</default>
+      <default>'#000000'</default>
       <summary>Default color of terminal background</summary>
       <description>Default color of terminal background, as a color specification (can be HTML-style hex digits, or a color name such as "red").</description>
     </key>
@@ -264,7 +264,7 @@
       <description>Sets what code the delete key generates. Possible values are "ascii-del" for the ASCII DEL character, "control-h" for Control-H (AKA the ASCII BS character), "escape-sequence" for the escape sequence typically  bound to backspace or delete. "escape-sequence" is normally considered the correct setting for the Delete key.</description>
     </key>
     <key name="use-theme-colors" type="b">
-      <default>true</default>
+      <default>false</default>
       <summary>Whether to use the colors from the theme for the terminal widget</summary>
       <description>If true, the theme color scheme used for text entry boxes will be used for the terminal, instead of colors provided by the user.</description>
     </key>
diff --git a/help/Makefile.am b/help/Makefile.am
index 057cdff..04177ed 100644
--- a/help/Makefile.am
+++ b/help/Makefile.am
@@ -30,7 +30,7 @@ HELP_MEDIA = figures/mate-terminal-default.png \
 #	      figures/terminal-window.png   This image is not used.
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS =
+IGNORE_HELP_LINGUAS = ja
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
