diff --git a/Makefile.in b/Makefile.in
index 05bdb6d..c48aaa9 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -15240,12 +15240,12 @@ uninstall-man: uninstall-man1 uninstall-man3 uninstall-man5 \
 @BUILD_LIBFDISK_TRUE@uninstall-hook-libfdisk:
 @BUILD_LIBFDISK_TRUE@	rm -f $(DESTDIR)$(libdir)/libfdisk.so*
 @BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@install-exec-hook-wall::
-@BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	chgrp tty $(DESTDIR)$(usrbin_execdir)/wall
-@BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	chmod g+s $(DESTDIR)$(usrbin_execdir)/wall
+@BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	echo chgrp tty $(DESTDIR)$(usrbin_execdir)/wall
+@BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	echo chmod g+s $(DESTDIR)$(usrbin_execdir)/wall
 
 @BUILD_WRITE_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@install-exec-hook-write::
-@BUILD_WRITE_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	chgrp tty $(DESTDIR)$(usrbin_execdir)/write
-@BUILD_WRITE_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	chmod g+s $(DESTDIR)$(usrbin_execdir)/write
+@BUILD_WRITE_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	echo chgrp tty $(DESTDIR)$(usrbin_execdir)/write
+@BUILD_WRITE_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	echo chmod g+s $(DESTDIR)$(usrbin_execdir)/write
 
 @BUILD_LAST_TRUE@install-exec-hook-last:
 @BUILD_LAST_TRUE@	cd $(DESTDIR)$(usrbin_execdir) && ln -sf last lastb
@@ -15253,8 +15253,8 @@ uninstall-man: uninstall-man1 uninstall-man3 uninstall-man5 \
 @BUILD_VIPW_TRUE@	cd $(DESTDIR)$(usrsbin_execdir) && ln -sf vipw vigr
 
 install-exec-hook:
-@BUILD_SU_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@	chown root:root $(DESTDIR)$(bindir)/su
-@BUILD_SU_TRUE@@MAKEINSTALL_DO_SETUID_TRUE@	chmod 4755 $(DESTDIR)$(bindir)/su
+@BUILD_SU_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@	echo chown root:root $(DESTDIR)$(bindir)/su
+@BUILD_SU_TRUE@@MAKEINSTALL_DO_SETUID_TRUE@	echo chmod 4755 $(DESTDIR)$(bindir)/su
 @BUILD_VIPW_TRUE@	cd $(DESTDIR)$(usrsbin_execdir) && ln -sf vipw vigr
 
 @BUILD_SETARCH_TRUE@$(SETARCH_MAN_LINKS):
@@ -15272,10 +15272,10 @@ install-exec-hook:
 @BUILD_SETARCH_TRUE@	done
 
 @BUILD_MOUNT_TRUE@install-exec-hook-mount:
-@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@	chown root:root $(DESTDIR)$(bindir)/mount
-@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_SETUID_TRUE@	chmod 4755 $(DESTDIR)$(bindir)/mount
-@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@	chown root:root $(DESTDIR)$(bindir)/umount
-@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_SETUID_TRUE@	chmod 4755 $(DESTDIR)$(bindir)/umount
+@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@	echo chown root:root $(DESTDIR)$(bindir)/mount
+@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_SETUID_TRUE@	echo chmod 4755 $(DESTDIR)$(bindir)/mount
+@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@	echo chown root:root $(DESTDIR)$(bindir)/umount
+@BUILD_MOUNT_TRUE@@MAKEINSTALL_DO_SETUID_TRUE@	echo chmod 4755 $(DESTDIR)$(bindir)/umount
 
 @BUILD_BASH_COMPLETION_TRUE@@BUILD_RUNUSER_TRUE@install-data-hook-bashcomp-runuser::
 @BUILD_BASH_COMPLETION_TRUE@@BUILD_RUNUSER_TRUE@@BUILD_SU_TRUE@	ln -sf su $(DESTDIR)$(bashcompletiondir)/runuser
