diff --git a/src/helpers/utf8.h b/src/helpers/utf8.h
index 90564ff..d74e105 100644
--- a/src/helpers/utf8.h
+++ b/src/helpers/utf8.h
@@ -18,8 +18,8 @@ along with XMOTO; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 =============================================================================*/
 
-#ifndef __UTF8_H__
-#define __UTF8_H__
+#ifndef __UTF88_H__
+#define __UTF88_H__
 
 #include <vector>
 #include <string>
