diff --git a/main.cpp b/main.cpp
index f8458b3..e3c737c 100644
--- a/source/main.cpp
+++ b/source/main.cpp
@@ -499,21 +499,6 @@ extern "C" __attribute__((visibility ("default"))) int main(int argc, char* argv
 
 extern "C" int main(int argc, char* argv[])
 {
-#if OS_UNIX
-	// Don't allow people to run the game with root permissions,
-	//	because bad things can happen, check before we do anything
-	if (geteuid() == 0)
-	{
-		std::cerr << "********************************************************\n"
-				  << "WARNING: Attempted to run the game with root permission!\n"
-				  << "This is not allowed because it can alter home directory \n"
-				  << "permissions and opens your system to vulnerabilities.   \n"
-				  << "(You received this message because you were either      \n"
-				  <<"  logged in as root or used e.g. the 'sudo' command.) \n"
-				  << "********************************************************\n\n";
-		return EXIT_FAILURE;
-	}
-#endif // OS_UNIX
 
 	EarlyInit();	// must come at beginning of main
 
