diff -ur lmms-0.4.12/CMakeLists.txt lmms-0.4.12.fix/CMakeLists.txt
--- lmms-0.4.12/CMakeLists.txt	2011-07-02 15:14:01.000000000 +0200
+++ lmms-0.4.12.fix/CMakeLists.txt	2012-01-07 16:13:07.000000000 +0100
@@ -356,7 +356,7 @@
 # set compiler flags
 #SET(CMAKE_BUILD_TYPE relwithdebug)
 SET(CMAKE_C_FLAGS "-O2 -g -Wall ${CMAKE_C_FLAGS}")
-SET(CMAKE_CXX_FLAGS "-O2 -g -fno-exceptions -Wall ${CMAKE_CXX_FLAGS}")
+SET(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -O2 -g -fno-exceptions -Wall ${CMAKE_CXX_FLAGS}")
 #SET(CMAKE_C_FLAGS_${CMAKE_BUILD_TYPE} "${CMAKE_C_FLAGS}")
 #SET(CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE} "${CMAKE_CXX_FLAGS}")
