diff -ur bzip2-1.0.6.orig//Makefile bzip2-1.0.6/Makefile
--- bzip2-1.0.6.orig//Makefile	2010-09-11 00:46:02.000000000 +0200
+++ bzip2-1.0.6/Makefile	2012-10-24 10:27:42.000000000 +0200
@@ -21,7 +21,7 @@
 LDFLAGS=
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS=-Wall -fPIC -Winline -O2 -g $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
 PREFIX=/usr/local
