diff -rupN gr0_15.orig/gunroar.desktop gr0_15/gunroar.desktop
--- gr0_15.orig/gunroar.desktop	1970-01-01 01:00:00.000000000 +0100
+++ gr0_15/gunroar.desktop	2014-01-08 14:33:42.016480184 +0100
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=gunroar
+Comment=gunroar
+Comment[fr]=gunroar
+Exec=gunroar -window
+Icon=gunroar
+Terminal=false
+Type=Application
+Categories=Game;SportsGame;
diff -rupN gr0_15.orig/Makefile gr0_15/Makefile
--- gr0_15.orig/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ gr0_15/Makefile	2014-01-08 14:41:55.252485011 +0100
@@ -0,0 +1,7 @@
+all:
+	echo "run make install"
+install:
+	install -d $(MYPREFIX)/share/applications
+	install -d $(MYPREFIX)/bin
+	cp -fR * $(MYPREFIX)/bin
+	cp gunroar.desktop $(MYPREFIX)/share/applications/	
