diff -Naur inetutils-1.9.orig/ifconfig/system/linux.c inetutils-1.9/ifconfig/system/linux.c
--- inetutils-1.9.orig/ifconfig/system/linux.c	2012-01-04 16:31:36.000000000 -0500
+++ inetutils-1.9/ifconfig/system/linux.c	2012-01-04 16:40:53.000000000 -0500
@@ -49,6 +49,10 @@
 #include "../ifconfig.h"
 
 
+#ifndef PATH_PROCNET_DEV
+  #define PATH_PROCNET_DEV "/proc/net/dev"
+#endif
+
 /* ARPHRD stuff.  */
 
 static void
