diff --git a/Config.kmk b/Config.kmk
index cac581b..37b038e 100644
--- a/Config.kmk
+++ b/Config.kmk
@@ -1499,7 +1499,7 @@ ASTOOL = $(VBOX_ASTOOL)
 #
 # Note! The __SOMETHING__ #defines are deprecated as they violate the C/C++ specs.
 #
-INCS += $(PATH_ROOT)/include $(PATH_OUT)
+INCS +=  ${HOME}/qt5/include  ${HOME}/${ARCH}/include $(PATH_ROOT)/include $(PATH_OUT)
 DEFS += VBOX
 DEFS.debug      := DEBUG DEBUG_$(subst $(subst _, ,_),_,$(USERNAME)) DEBUG_USERNAME=$(subst $(subst _, ,_),_,$(USERNAME))
 DEFS.dbgopt      = $(DEFS.debug)
@@ -2022,9 +2022,9 @@ endif
 
 # The X11 LIBPATH. (The amd64 hack is for RHEL4.)
 ifeq ($(KBUILD_TARGET_ARCH),amd64)
- VBOX_LIBPATH_X11 ?= /usr/X11R6/lib64 /usr/X11R6/lib
+ VBOX_LIBPATH_X11 ?= ${HOME}/${ARCH}/lib
 else
- VBOX_LIBPATH_X11 ?= /usr/X11R6/lib
+ VBOX_LIBPATH_X11 ?= ${HOME}/${ARCH}/lib
 endif
 
 # 32bit X11 LIBPATH for building 32bit guest additions
@@ -4549,7 +4549,7 @@ TEMPLATE_VBOXR3EXE_LDFLAGS.x86         = -m32
 TEMPLATE_VBOXR3EXE_LDFLAGS.amd64       = -m64
 TEMPLATE_VBOXR3EXE_LDFLAGS.sparc32     = -m32
 TEMPLATE_VBOXR3EXE_LDFLAGS.sparc64     = -m64
-TEMPLATE_VBOXR3EXE_LDFLAGS.linux       = -Wl,-z,noexecstack,-z,relro $(VBOX_LD_as_needed)
+TEMPLATE_VBOXR3EXE_LDFLAGS.linux       = -Wl,-z,noexecstack,-z,relro $(VBOX_LD_as_needed) -L${HOME}/${ARCH}/lib
 TEMPLATE_VBOXR3EXE_LDFLAGS.solaris     = -Wl,-z,ignore # same as VBOX_LD_as_needed
 
  ifeq ($(KBUILD_TARGET),linux)
@@ -5351,7 +5351,7 @@ TEMPLATE_VBOXMAINEXE_LDFLAGS.x86         = -m32
 TEMPLATE_VBOXMAINEXE_LDFLAGS.amd64       = -m64
  ifeq ($(KBUILD_TARGET),linux)
 TEMPLATE_VBOXMAINEXE_LIBS                = pthread m rt $(LIB_RUNTIME) dl
-TEMPLATE_VBOXMAINEXE_LDFLAGS.linux       = -Wl,-z,noexecstack,-z,relro $(VBOX_LD_as_needed)
+TEMPLATE_VBOXMAINEXE_LDFLAGS.linux       = -Wl,-z,noexecstack,-z,relro $(VBOX_LD_as_needed) -L${HOME}/${ARCH}/lib
  else ifeq ($(KBUILD_TARGET),os2)
 TEMPLATE_VBOXMAINEXE_TOOL                = GXX3OMF
 TEMPLATE_VBOXMAINEXE_LIBS                = $(LIB_RUNTIME)
@@ -6949,7 +6949,7 @@ else # !darwin
    # Add correct detection for you distro after the /usr/../java-6-sun line.
    VBOX_JAVA_HOME := $(firstword $(wildcard \
 	$(KBUILD_DEVTOOLS)/linux.x86/jdk1.6.0_27 \
-	/usr/lib/jvm/java-6-sun  \
+	$(HOME)/$(ARCH)  \
 	/usr/lib/jvm/java-6-openjdk \
 	/usr/lib/jvm/java-6-openjdk-i386 \
 	/usr/lib/jvm/java-6-openjdk-amd64 \
diff --git a/configure b/configure
index eabd710..c0f7134 100755
--- a/configure
+++ b/configure
@@ -20,7 +20,7 @@ export LC_ALL
 # append some extra paths
 PATH="$PATH:/opt/gnome/bin"
 # Solaris (order of paths important for tr, grep, sed to work)
-PATH="/usr/xpg4/bin:$PATH:/usr/local/bin:/usr/sfw/bin:/usr/ccs/bin"
+PATH="/usr/xpg4/bin:$PATH:${HOME}/${ARCH}/bin:/usr/sfw/bin:/usr/ccs/bin"
 ORGPATH=$PATH
 
 # Wrapper for ancient /usr/bin/which on darwin that always returns 0
@@ -117,13 +117,13 @@ XSLTPROC="xsltproc"
 GENISOIMAGE="genisoimage"
 MKISOFS="mkisofs"
 INCCRYPTO=""
-LIBCRYPTO="-lssl -lcrypto"
+LIBCRYPTO="-lssl -lcrypto -L${HOME}/${ARCH}/lib -I${HOME}/${ARCH}/include  "
 LIBPTHREAD="-lpthread"
-LIBCAP="-lcap"
+LIBCAP="-lcap -L${HOME}/${ARCH}/lib -I${HOME}/${ARCH}/include"
 GSOAP=""
 GSOAP_IMPORT=""
-INCX11="/usr/local/include"
-LIBX11="-L/usr/X11R6/lib -L/usr/X11R6/lib64 -L/usr/local/lib -lXext -lX11"
+INCX11="${HOME}/${ARCH}/include"
+LIBX11="-L${HOME}/${ARCH}/lib -L${HOME}/${ARCH}/lib64 -L${HOME}/${ARCH}/lib -lXext -lX11"
 LIBXCURSOR="-lXcursor"
 LIBXMU="-lXmu"
 LIBXINERAMA="-lXinerama"
@@ -132,31 +132,33 @@ MAKESELF="makeself"
 MESA="-lGL"
 INCZ=""
 LIBZ="-lz"
-INCVNCSERVER=""
-LIBVNCSERVER="-lvncserver"
-INCDEVMAPPER=""
-LIBDEVMAPPER="-ldevmapper"
+INCVNCSERVER="-I${HOME}/${ARCH}/include"
+LIBVNCSERVER="-L${HOME}/${ARCH}/lib -lvncserver"
+INCDEVMAPPER="-I${HOME}/${ARCH}/include"
+LIBDEVMAPPER="-L${HOME}/${ARCH}/lib -ldevmapper"
 CXX_FLAGS=""
 if [ "$OS" = "freebsd" ]; then
-  INCCURL="-I/usr/local/include"
-  LIBCURL="-L/usr/local/lib -lcurl"
-  INCPULSE="-I/usr/local/include"
-  LIBPULSE="-L/usr/local/lib"
-  INCPNG="-I/usr/local/include"
-  LIBPNG="-L/usr/local/lib -lpng"
+  INCCURL="-I${HOME}/${ARCH}/include"
+  LIBCURL="-L${HOME}/${ARCH}/lib -lcurl"
+  INCPULSE="-I${HOME}/${ARCH}/include"
+  LIBPULSE="-L${HOME}/${ARCH}/lib"
+  INCPNG="-I${HOME}/${ARCH}/include"
+  LIBPNG="-L${HOME}/${ARCH}/lib -lpng"
 else
-  INCCURL=""
-  LIBCURL="-lcurl"
-  INCPNG=""
-  LIBPNG="-lpng"
+  INCCURL="-I${HOME}/${ARCH}/include "
+  INCPULSE="-I${HOME}/${ARCH}/include "
+  LIBCURL="-lcurl -L${HOME}/${ARCH}/lib"
+  LIBPULSE="-lcurl -L${HOME}/${ARCH}/lib"
+  INCPNG="-I${HOME}/${ARCH}/include"
+  LIBPNG="-lpng  -L${HOME}/${ARCH}/lib"
 fi
 INCVPX=""
 LIBVPX="-lvpx"
 PKGCONFIG="`which_wrapper pkg-config`"
-PYTHONDIR="/usr /usr/local"
-QT4DIR="/usr/lib/qt4 /usr/share/qt4 /usr/lib64/qt4 /usr /usr/local"
+PYTHONDIR="/usr ${HOME}/${ARCH}"
+QT4DIR="/usr/lib/qt4 /usr/share/qt4 /usr/lib64/qt4 /usr ${HOME}/${ARCH}"
 QT4DIR_PKGCONFIG=1
-QT5DIR="/usr/lib/qt5 /usr/share/qt5 /usr/lib64/qt5 /usr /usr/local"
+QT5DIR="${HOME}/qt5 /usr/share/qt5 /usr/lib64/qt5 /usr ${HOME}/${ARCH}"
 QT5DIR_PKGCONFIG=1
 KBUILDDIR="`cd \`dirname $0\`; pwd`/kBuild"
 DEVDIR="`cd \`dirname $0\`; pwd`/tools"
@@ -1231,7 +1233,7 @@ extern "C" int main(void)
 #endif
 }
 EOF
-  if test_compile "-lasound" asound asound; then
+  if test_compile "-lasound -L${HOME}/${ARCH}/lib -I${HOME}/${ARCH}/include" asound asound; then
     test_execute
   fi
 }
diff --git a/kBuild/sdks/LIBSDL.kmk b/kBuild/sdks/LIBSDL.kmk
index dbec4c2..e4ae38a 100644
--- a/kBuild/sdks/LIBSDL.kmk
+++ b/kBuild/sdks/LIBSDL.kmk
@@ -55,7 +55,7 @@ ifndef PATH_SDK_LIBSDL
     _SDK_LIBSDK_IS_FRAMEWORK := yes
    endif
   endif
-
+  PATH_SDK_LIBSDL=/usr
   ifeq ($(PATH_SDK_LIBSDL),)
    PATH_SDK_LIBSDL := $(firstword $(foreach root,$(KBUILD_LIB_SEARCH_ROOTS) \
    	,$(if $(wildcard $(addsuffix libSDL.*,$(addprefix $(root),$(KBUILD_LIB_SEARCH_SUBS)))),$(root),) ))
diff --git a/src/VBox/Additions/linux/Makefile b/src/VBox/Additions/linux/Makefile
index 5cf091c..3238981 100644
--- a/src/VBox/Additions/linux/Makefile
+++ b/src/VBox/Additions/linux/Makefile
@@ -28,11 +28,11 @@ KBUILD_VERBOSE =
 
 all:
 	@echo "=== Building 'vboxguest' module ==="
-	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxguest
+	@echo $(MAKE) KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3)/ KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxguest
 	@if [ -f vboxguest/vboxguest.ko ]; then \
-	    cp vboxguest/vboxguest.ko .; \
+	    echo cp vboxguest/vboxguest.ko .; \
 	 else \
-	    cp vboxguest/vboxguest.o .; \
+	    echo cp vboxguest/vboxguest.o .; \
 	 fi
 	@echo
 	@if [ -d vboxsf ]; then \
@@ -40,11 +40,11 @@ all:
 	        cp vboxguest/Module.symvers vboxsf; \
 	    fi; \
 	    echo "=== Building 'vboxsf' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxsf || exit 1; \
+	    echo $(MAKE) KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3)/ KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxsf || exit 1; \
 	    if [ -f vboxsf/vboxsf.ko ]; then \
-	        cp vboxsf/vboxsf.ko .; \
+	        echo cp vboxsf/vboxsf.ko .; \
 	    else \
-	        cp vboxsf/vboxsf.o .; \
+	        echo cp vboxsf/vboxsf.o .; \
 	    fi; \
 	    echo; \
 	fi
@@ -53,7 +53,7 @@ all:
 	        cp vboxguest/Module.symvers vboxvideo; \
 	    fi; \
 	    echo "=== Building 'vboxvideo' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxvideo || \
+	    echo $(MAKE) KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3)/ KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxvideo || \
 	      (echo; echo "Building of vboxvideo failed, ignoring!"; echo); \
 	    if [ -f vboxvideo/vboxvideo.ko ]; then \
 	        cp vboxvideo/vboxvideo.ko .; \
@@ -64,23 +64,8 @@ all:
 	fi
 
 install:
-	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxguest install
-	@if [ -d vboxsf ]; then \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxsf install; \
-	fi
-	@if [ -d vboxvideo ]; then \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxvideo install; \
-	fi
 
 clean:
-	@$(MAKE) -C vboxguest clean
-	@if [ -d vboxsf ]; then \
-	    $(MAKE) -C vboxsf clean; \
-	fi
-	@if [ -d vboxvideo ]; then \
-	    $(MAKE) -C vboxvideo clean; \
-	fi
-	rm -f vboxguest.*o vboxsf.*o vboxvideo.*o
 
 check:
 	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxguest check
diff --git a/src/VBox/HostDrivers/Makefile.kmk b/src/VBox/HostDrivers/Makefile.kmk
index 0647806..fd45f85 100644
--- a/src/VBox/HostDrivers/Makefile.kmk
+++ b/src/VBox/HostDrivers/Makefile.kmk
@@ -59,7 +59,7 @@ endif # !defined(VBOX_ONLY_DOCS) && !defined(VBOX_ONLY_VALIDATIONKIT)
 #
 # Build test for the linux host kernel modules.
 #
-if   defined(VBOX_WITH_TESTCASES) \
+if   defined(VVVBOX_WITH_TESTCASES) \
  && !defined(VBOX_ONLY_SDK) \
  && !defined(VBOX_ONLY_DOCS) \
  && !defined(VBOX_ONLY_VALIDATIONKIT) \
diff --git a/src/VBox/HostDrivers/linux/Makefile b/src/VBox/HostDrivers/linux/Makefile
index 85466b1..d3a5f8a 100644
--- a/src/VBox/HostDrivers/linux/Makefile
+++ b/src/VBox/HostDrivers/linux/Makefile
@@ -38,7 +38,7 @@ KBUILD_VERBOSE =
 
 all:
 	@echo "=== Building 'vboxdrv' module ==="
-	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxdrv
+	@make  KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3)  KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxdrv
 	@cp vboxdrv/vboxdrv.ko .
 	@echo
 	@if [ -d vboxnetflt ]; then \
@@ -46,8 +46,8 @@ all:
 	        cp vboxdrv/Module.symvers vboxnetflt; \
 	    fi; \
 	    echo "=== Building 'vboxnetflt' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxnetflt || exit 1; \
-	    cp vboxnetflt/vboxnetflt.ko .; \
+	    echo make  KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxnetflt || exit 1; \
+	    echo cp vboxnetflt/vboxnetflt.ko .; \
 	    echo; \
 	fi
 	@if [ -d vboxnetadp ]; then \
@@ -55,8 +55,8 @@ all:
 	        cp vboxdrv/Module.symvers vboxnetadp; \
 	    fi; \
 	    echo "=== Building 'vboxnetadp' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxnetadp || exit 1; \
-	    cp vboxnetadp/vboxnetadp.ko .; \
+	    echo make  KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxnetadp || exit 1; \
+	    echo cp vboxnetadp/vboxnetadp.ko .; \
 	    echo; \
 	fi
 	@if [ -d vboxpci ]; then \
@@ -64,35 +64,14 @@ all:
 	        cp vboxdrv/Module.symvers vboxpci; \
 	    fi; \
 	    echo "=== Building 'vboxpci' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxpci || exit 1; \
-	    cp vboxpci/vboxpci.ko .; \
+	    echo make  KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxpci || exit 1; \
+	    echo cp vboxpci/vboxpci.ko .; \
 	    echo; \
 	fi
 
 install:
-	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxdrv install
-	@if [ -d vboxnetflt ]; then \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxnetflt install; \
-	fi
-	@if [ -d vboxnetadp ]; then \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxnetadp install; \
-	fi
-	@if [ -d vboxpci ]; then \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxpci install; \
-	fi
 
 clean:
-	@$(MAKE) -C vboxdrv clean
-	@if [ -d vboxnetflt ]; then \
-	    $(MAKE) -C vboxnetflt clean; \
-	fi
-	@if [ -d vboxnetadp ]; then \
-	    $(MAKE) -C vboxnetadp clean; \
-	fi
-	@if [ -d vboxpci ]; then \
-	    $(MAKE) -C vboxpci clean; \
-	fi
-	rm -f vboxdrv.ko vboxnetflt.ko vboxnetadp.ko vboxpci.ko 
 
 check:
 	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxdrv check
diff --git a/src/VBox/Installer/linux/Makefile.include.footer b/src/VBox/Installer/linux/Makefile.include.footer
index 7830bfd..6b9694e 100644
--- a/src/VBox/Installer/linux/Makefile.include.footer
+++ b/src/VBox/Installer/linux/Makefile.include.footer
@@ -82,10 +82,10 @@ endif
 
 # OL/UEK: disable module signing for external modules -- we don't have any private key
 $(MODULE):
-	$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) CONFIG_MODULE_SIG= -C $(KERN_DIR) SUBDIRS=$(CURDIR) SRCROOT=$(CURDIR) -j$(JOBS) modules
+	echo make KBUILD_VERBOSE=$(KBUILD_VERBOSE) KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3)/ CONFIG_MODULE_SIG= -C $(KERN_DIR) SUBDIRS=$(CURDIR) SRCROOT=$(CURDIR) -j$(JOBS) modules
 
 modules_install:
-	$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) CONFIG_MODULE_SIG= -C $(KERN_DIR) SUBDIRS=$(CURDIR) SRCROOT=$(CURDIR) modules_install
+	echo make KBUILD_VERBOSE=$(KBUILD_VERBOSE) KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3)/ CONFIG_MODULE_SIG= -C $(KERN_DIR) SUBDIRS=$(CURDIR) SRCROOT=$(CURDIR) modules_install
 
 endif
 
diff --git a/src/VBox/Installer/linux/Makefile.include.header b/src/VBox/Installer/linux/Makefile.include.header
index bda987a..5d6c7e0 100644
--- a/src/VBox/Installer/linux/Makefile.include.header
+++ b/src/VBox/Installer/linux/Makefile.include.header
@@ -107,7 +107,7 @@ ifeq ($(KERNELRELEASE),)
  ifndef KERN_DIR
   KERN_DIR := /lib/modules/$(KERN_VER)/build
   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-   KERN_DIR := /usr/src/linux
+   KERN_DIR := KERN_DIR=$(HOME)/ydfs/build/linux-x86_64-$(KERNEL3)/
    ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
     $(error Error: unable to find the sources of your current Linux kernel. \
 	           Specify KERN_DIR=<directory> and run Make again)
