diff --git a/third_party/re2/src/re2/filtered_re2.h b/third_party/re2/src/re2/filtered_re2.h
index b1317cc..965b5c9 100644
--- a/third_party/re2/src/re2/filtered_re2.h
+++ b/third_party/re2/src/re2/filtered_re2.h
@@ -49,7 +49,7 @@ class FilteredRE2 {
   // the search text should be lowercased first to find matching
   // strings from the set of strings returned by Compile.  Call after
   // all Add calls are done.
-  void Compile(std::vector<string>* strings_to_match);
+  void Compile(std::vector<std::string>* strings_to_match);
 
   // Returns the index of the first matching regexp.
   // Returns -1 on no match. Can be called prior to Compile.
