# Debian 6 has autoconf (GNU Autoconf) 2.67
diff -ur lxsession.orig//configure.ac lxsession/configure.ac
--- lxsession.orig//configure.ac	2013-09-07 05:53:37.000000000 +0200
+++ lxsession/configure.ac	2013-09-07 05:53:55.000000000 +0200
@@ -2,7 +2,7 @@
 
 # Autotols programs
 AM_INIT_AUTOMAKE([1.11.1 -Wall -Werror foreign subdir-objects])
-AC_PREREQ([2.68])
+AC_PREREQ([2.67])
 IT_PROG_INTLTOOL([0.40.0],[no-xml])
 
 #Initialize libtool
