diff --git a/client/mysql.cc b/client/mysql.cc
index dcd9dfe..13a0877 100644
--- a/client/mysql.cc
+++ b/client/mysql.cc
@@ -61,9 +61,9 @@ static char *server_version= NULL;
 #include "client_metadata.h"
 
 extern "C" {
-#if defined(HAVE_CURSES_H) && defined(HAVE_TERM_H)
 #include <curses.h>
 #include <term.h>
+#if defined(HAVE_CURSES_H) && defined(HAVE_TERM_H)
 #else
 #if defined(HAVE_TERMIOS_H)
 #include <termios.h>
